common/src/main/java/tds/common/cache/RedisJsonSerializer.java

```diff
@@ -0,0 +1,53 @@
+package tds.common.cache;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.SerializationException;
+
+import java.io.IOException;
+
+/**
+ * This Redis serializer uses an ObjectMapper to serialize objects.
+ */
+public class RedisJsonSerializer implements RedisSerializer<Object> {
+    private final static Logger LOG = LoggerFactory.getLogger(RedisJsonSerializer.class);
+
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Constructor
+     *
+     * @param objectMapper The application ObjectMapper
+     */
+    public RedisJsonSerializer(final ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper.copy();
+        this.objectMapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
```

Comment:
When serializing/deserializing to/from Redis, we don't know the class of Object that we're serializing.  I'm copying the existing ObjectMapper and telling it to embed class information in the serialized value.